File: src/main/java/com/sstep/recipe/manager/dto/RecipeData.java
package com.sstep.recipe.manager.dto;

import com.sstep.recipe.manager.vo.IngredientVo;
import com.sstep.recipe.manager.vo.RecipeStepVo;
import com.sstep.recipe.manager.vo.RecipeVo;

import java.util.List;


public record RecipeData(
        List<IngredientVo> ingredients,
        List<RecipeStepVo> steps
) {

    public RecipeData() {
        this(List.of(), List.of());
    }

    public RecipeData(final RecipeVo recipeVo) {
        this(recipeVo.getIngredients(), recipeVo.getSteps());
    }
}

File: src/main/kotlin/com/sstep/recipe/manager/dto/RecipeData.kt
package com.sstep.recipe.manager.dto

import com.fasterxml.jackson.annotation.JsonCreator
import com.sstep.recipe.manager.vo.IngredientVo
import com.sstep.recipe.manager.vo.RecipeStepVo
import com.sstep.recipe.manager.vo.RecipeVo

data class RecipeData @JsonCreator constructor(
    val ingredients: List<IngredientVo> = emptyList(),
    val steps: List<RecipeStepVo> = emptyList(),
) {
    constructor(recipeVo: RecipeVo) : this(recipeVo.ingredients, recipeVo.steps)
}